hydra/utils.py

```diff
@@ -34,8 +36,10 @@ def call(config: Union[ObjectConf, DictConfig], *args: Any, **kwargs: Any) -> An
         else:
             assert callable(type_or_callable)
             return _call_callable(type_or_callable, config, *args, **kwargs)
+    except ValueError as e:
+        raise HydraException(f"Error calling function : {e}") from e
     except Exception as e:
-        raise HydraException(f"Error instantiating '{cls}' : {e}") from e
+        raise HydraException(f"Error calling '{cls}' : {e}") from e
```

Comment:
Differentiating between the two errors based on these exceptions is very fragile.
I suggest that you put the exception re-raising in call and instantiate, and have them both call a common private function. This way you know if this is a call or instantiating based on the context.

Reply:
ohh just to give some context, these are just different kind of exceptions - missing `target` key in config, vs exception. I think its better if always raise a `HydraException` (and use it to wrap a `ValueError`). Should we make a new `HydraException` class (that extends `ValueError`) ?

Reply:
Who is throwing a ValueError on a missing target field?
You realize that ValueError is very common any instantiated code is likely to throw it too?

Reply:
Now we throw a HydraException if the `target` key is missing. Now might be a good time to have a `MissingKey` exception for Hydra, along the lines of `MissingConfig` exception. The `MissingKey` exception can be used by HPO-plugins as well.

Reply:
OmegaConf got that, no need to replicate.
What happens if you just access target blindly?

Reply:
Used to raise a `ValueError`

Reply:
> Who is throwing a ValueError on a missing target field?

Reply:
https://github.com/facebookresearch/hydra/blob/master/hydra/_internal/utils.py#L561 . Last changed in https://github.com/facebookresearch/hydra/commit/a32958846d488aa85d324b11a3f2ac23a0fcbda2 when deprecating `cls`.

Reply:
and if we just make it a HydraException there?

Reply:
Yeah did that, based on your feedback: https://github.com/facebookresearch/hydra/pull/766/commits/54107b91349173058cd1c45d3200863dc43e5b23